app/javascript/packages/prompt-on-navigate/index.ts

```diff
@@ -0,0 +1,66 @@
+import { trackEvent } from '@18f/identity-analytics';
+
+export type PromptOnNavigateOptions = {
+  stillOnPageIntervalsInSeconds: number[];
+};
+
+const defaults = {
+  stillOnPageIntervalsInSeconds: [5, 15, 30],
+};
+
+/**
+ * Configures the window.onbeforeunload handler such that the user will be prompted before
+ * reloading or navigating away
+ * @param options {PromptOnNavigateOptions}
+ * @returns {() => void} A function that, when called, will "clean up" -- restore the prior onbeforeunload handler and cancel any pending timeouts.
+ */
+export function promptOnNavigate(options: PromptOnNavigateOptions = defaults): () => void {
+  let stillOnPageTimer: NodeJS.Timeout | undefined;
```

Comment:
This always gets confusing for me, because I think `window.setTimeout` return value is a number in TypeScript, and `setTimeout` return value is this `NodeJS.Timeout`, but the latter only applies in Node.js, not in the browser. And honestly, I couldn't tell you off the top of my head if `setTimeout` calls Node's version or JSDOM's window-proxied version when called in our Mocha environment.

Maybe it'd be best to just call them as `window.setTimeout` and `window.clearTimeout` to be explicit about what we're expecting?

Reply:
Yeah that's always bugged me too. I'll update to `number` and `window.setTimeout` / `window.clearTimeout`